tutorials/20_REST_API.ipynb

```diff
@@ -0,0 +1,538 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Tutorial: Using Haystack with REST API\n",
+    "\n",
+    "- **Level**: Advanced\n",
+    "- **Time to complete**: 30 minutes\n",
+    "- **Prerequisites**: Basic understanding of Docker, basic knowledge of Haystack pipelines \n",
+    "- **Nodes Used**: `ElasticsearchDocumentStore`, `EmbeddingRetriever`\n",
+    "- **Goal**: Learn how you can interact with Haystack through REST API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "\n",
+    "Haystack enables you to apply the latest NLP technology to your own data and create production-ready applications. Building an end-to-end NLP application requires the combination of multiple concepts. Here are those consepts:\n",
+    "* **DocumentStore** stores the data. You will use Elasticsearch for this tutorial.\n",
+    "* **Haystack** pipelines convert documents, index them to DocumentStore, and run NLP tasks such as question answering and document search.\n",
+    "* **REST API** interacts with Haystack and other applications to pass query and response between them.\n",
+    "* **Docker** simplifies the environment set-up needed to have Elasticsearch running.\n",
+    "\n",
+    "This tutorial introduces you to all the concepts needed to build an end-to-end document search application. After completing this tutorial, you will have learned how to create a pipeline YAML file, index files and query your application using REST API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preparing the Environment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Update or install Docker and Docker Compose, then launch Docker:\n",
+    "\n",
+    "Set up Docker to start an Elasticsearch container. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "apt-get update && apt-get install docker && apt-get install docker-compose\n",
+    "service docker start"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. Install Haystack:\n",
+    "\n",
+    "Install the latest version of Haystack from the main branch and all its dependencies. REST API and `xpdf` are also required.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash \n",
+    "\n",
+    "pip install --upgrade pip\n",
+    "pip install 'farm-haystack[all]'\n",
+    "pip install -e rest_api/\n",
+    "\n",
+    "brew install xpdf # required for PDFToTextConverter node"
```

Comment:
Sorry for the intrusion... by the way, great job!

I think that `brew` is installed by default only in macOS, not in Linux and Windows.

From [Haystack workflows for tests](https://github.com/deepset-ai/haystack/blob/0e05f71f330ab0ad14541efa64f2ba95ce87c2e6/.github/workflows/tests.yml#L461), I see that you can install `xpdf` on Linux with the following command:
`wget --no-check-certificate https://dl.xpdfreader.com/xpdf-tools-linux-4.04.tar.gz && tar -xvf xpdf-tools-linux-4.04.tar.gz && sudo cp xpdf-tools-linux-4.04/bin64/pdftotext /usr/local/bin`

For Windows you can install `xpdf` with `choco install xpdf-utils`, but you need to have Chocolatey.